build-tools/automation/azure-pipelines.yaml

```diff
@@ -1208,11 +1208,23 @@ stages:
     - powershell: |
         # We require at a minimum version 2.8.0 of the azure cli installed for the ESRP signing
         # First, check for what version is currently installed on the machine
+        $ErrorActionPreference = "Continue"
         & az --version
-        $Versions = & az --version 
-        $AzureVersion = $Versions[0]
-        $AzureVersionFields = $AzureVersion.split(" ", [StringSplitOptions]::RemoveEmptyEntries)
-        $CurrentVersion = New-Object -TypeName System.Version -ArgumentList $AzureVersionFields[1]
+
+        # Test whether az actually worked.  If it does not work, use a $CurrentVersion 
+        # of 0 in order to force us to install Azure CLI no matter what.
+        if ($?)
+        {
+            $Versions = & az --version 
+            $AzureVersion = $Versions[0]
+            $AzureVersionFields = $AzureVersion.split(" ", [StringSplitOptions]::RemoveEmptyEntries)
+            $CurrentVersion = New-Object -TypeName System.Version -ArgumentList $AzureVersionFields[1]
+        }
+        else 
+        {
+            $CurrentVersion = New-Object -TypeName System.Version -ArgumentList "0.0"
+        }
+        
         $Version280 = New-Object -TypeName System.Version -ArgumentList "2.8.0"
```

Comment:
@trevors20: it occurs to me that *after* this point, we should "reset" [`$ErrorActionPreference`](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-7.1#erroractionpreference) back to `Stop` (which I assume is the default), so that if the subsequent `brew` commands fail (e.g. `brew` isn't installed either!), the step properly fails with an error, instead of blithely continuing execution…

Reply:
Great idea.  Let me do that.